typegraph/core/src/runtimes/prisma/relationship/discovery.rs

```diff
@@ -0,0 +1,303 @@
+// Copyright Metatype OÜ, licensed under the Mozilla Public License Version 2.0.
+// SPDX-License-Identifier: MPL-2.0
+
+use std::collections::HashSet;
+
+use crate::global_store::with_store;
+use crate::runtimes::prisma::type_utils::{self, has_fkey};
+use crate::types::Type;
+use crate::wit::core::TypeId;
+use crate::{errors::Result, types::Struct};
+
+use super::RelationshipRegistry;
+use super::{Cardinality, Relationship, RelationshipModel};
+
+/// scan model `model` for relationships
+pub fn scan_model(model: &Struct, registry: &RelationshipRegistry) -> Result<Vec<Relationship>> {
+    with_store(|s| {
+        let mut result = Vec::new();
+        let mut self_relationships = HashSet::new();
+
+        for (prop, ty) in model.data.props.iter() {
+            let ty = s.resolve_proxy(*ty)?;
+            // TODO why would it be possible to have self relationship?
+            if registry.has(ty, prop)
+            /* || self_relationships.contains(&ty) */
+            {
+                continue;
+            }
+
+            match s.get_type(ty)? {
+                Type::Struct(target_model) => {
```

Comment:
should not the runtime also be checked?

Reply:
We can expect it to have the same runtime. In a graph, child types always inherit the runtime of the parent, unless it is overridden by the runtime of the materializer of a function type. Here we have a struct, so surely has the same runtime as the parent.

Reply:
```
t.struct({
"id": t.string(),
"similarOnes": http.get(A, B)
})
```
is already supported and will happen more than though. Maybe the runtime segmentation already happen somewhere centralised but I am not clear enough on the new SDK.

Reply:
In that case, `http.get(...)` will return a `t.func`, hinting it would be run on a different runtime.